lib/drivers/bluetoothLE/bluetoothLEDriver.js

```diff
@@ -0,0 +1,196 @@
+/*
+ * == BSD2 LICENSE ==
+ * Copyright (c) 2019, Tidepool Project
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the associated License, which is identical to the BSD 2-Clause
+ * License as published by the Open Source Initiative at opensource.org.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the License for more details.
+ *
+ * You should have received a copy of the License along with this program; if
+ * not, you can obtain one from Tidepool Project at tidepool.org.
+ * == BSD2 LICENSE ==
+ */
+
+import _ from 'lodash';
+import sundial from 'sundial';
+
+import TZOUtil from '../../TimezoneOffsetUtil';
+import annotate from '../../eventAnnotations';
+import { remote } from 'electron';
+
+const isBrowser = typeof window !== 'undefined';
+// eslint-disable-next-line no-console
+const debug = isBrowser ? require('bows')('BluetoothLEDriver') : console.log;
+
+module.exports = (config) => {
+  const cfg = _.clone(config);
+  cfg.deviceTags = ['bgm'];
+
+  // With no date & time settings changes available,
+  // timezone is applied across-the-board
+  cfg.tzoUtil = new TZOUtil(cfg.timezone, new Date().toISOString(), []);
+
+  return {
+   /* eslint no-param-reassign:
+      [ "error", { "props": true, "ignorePropertyModificationsFor": ["data"] } ] */
+
+   detect(deviceInfo, cb) {
+     debug('no detect function needed', deviceInfo);
+     cb(null, deviceInfo);
+   },
+
+   setup(deviceInfo, progress, cb) {
+     debug('in setup!');
+     progress(100);
+     cb(null, { deviceInfo });
+   },
+
+   async connect(progress, data, cb) {
+     try {
+       debug('in connect!');
+       await cfg.deviceComms.ble.connectTimeout();
+     } catch (error) {
+       return cb(error, null);
+     }
+     return cb(null, data);
+   },
+
+   async getConfigInfo(progress, data, cb) {
+     debug('in getConfigInfo', data);
+     progress(0);
+
+     try {
+       _.assign(cfg.deviceInfo , await cfg.deviceComms.ble.getDeviceInfo());
+       cfg.deviceTags = ['bgm'];
+       cfg.deviceInfo.deviceId = `${[cfg.deviceInfo.manufacturers]}-${cfg.deviceInfo.model}-${remote.getGlobal('bluetoothDeviceId')}`;
+       cfg.builder.setDefaults({ deviceId: cfg.deviceInfo.deviceId });
+
+       cfg.deviceComms.ble.once('numberOfRecords', (result) => {
+         debug('Number of records:', result);
+       });
```

Comment:
As you most likely know, at some point this becomes meaningless. Should one get the number of records if that doesn't improve the program flow?

Reply:
You're right, I should remove this as I'm not using that value anymore.